src/components/VideoPlayer/VideoPlayerControls/index.tsx

```diff
@@ -56,9 +58,10 @@ function VideoPlayerControls({duration, position, url, videoPlayerRef, isPlaying
     };
 
     const enterFullScreenMode = useCallback(() => {
+        isFullscreenRef.current = true;
```

Comment:
Something like this?
```ts
    const setIsFullScreen = useCallback((isFullScreen: boolean) => {
        isFullScreenRef.current = isFullScreen;
    }, []);
```
Wouldn't it be too simple since we are dealing with ref that can be changed anytime?